web-bundle/src/main/resources/com/graphhopper/maps/js/main-template.js

```diff
@@ -115,72 +119,23 @@ $(document).ready(function (e) {
         cmEditor.cm.execCommand('selectAll');
         cmEditor.cm.refresh();
     }
-    $("#custom-model-example").click(function() {
+
+    $("#custom-model-example").click(function () {
         showCustomModelExample();
         return false;
     });
 
-    var sendCustomData = function() {
-       mapLayer.clearElevation();
-       mapLayer.clearLayers();
-       flagAll();
-
-       var infoDiv = $("#info");
-       infoDiv.empty();
-       infoDiv.show();
-       var routeResultsDiv = $("<div class='route_results'/>");
-       infoDiv.append(routeResultsDiv);
-       routeResultsDiv.html('<img src="img/indicator.gif"/> Search Route ...');
-
-       var points = [];
-       for(var idx = 0; idx < ghRequest.route.size(); idx++) {
-           var point = ghRequest.route.getIndex(idx);
-           if (point.isResolved()) {
-               points.push([point.lng, point.lat]);
-           } else {
-               routeResultsDiv.html("Unresolved points");
-               return;
-           }
-       }
-
-       var customModel = cmEditor.jsonObj;
-       if (customModel === null) {
-           routeResultsDiv.html("Invalid custom model");
-           return;
-       }
-
-       const details = cmEditor.getUsedCategories();
-       details.push('average_speed');
-       details.push('distance');
-       details.push('time');
-       var request = {
-           points: points,
-           points_encoded: false,
-           elevation: ghRequest.api_params.elevation,
-           profile: ghRequest.api_params.profile,
-           custom_model: customModel,
-           "ch.disable": true,
-           details: details
-       }
-
-       $.ajax({
-           url: host + "/route",
-           type: "POST",
-           contentType: 'application/json; charset=utf-8',
-           dataType: "json",
-           data: JSON.stringify(request),
-           success: createRouteCallback(ghRequest, routeResultsDiv, "", true),
-           error: function(err) {
-               routeResultsDiv.html("Error response: cannot process input");
-               var json = JSON.parse(err.responseText);
-               createRouteCallback(ghRequest, routeResultsDiv, "", true)(json);
-           }
-        });
+    var sendCustomData = function () {
+        ghRequest.ignoreCustomErrors = false;
+        mySubmit();
     };
 
-    // todo: prevent this as long as custom model is invalid?
-    // so far it is useful so we can send the request regardless of the invalidation
-    cmEditor.setExtraKey('Ctrl-Enter', sendCustomData);
+    var sendCustomDataIgnoreErrors = function () {
+        ghRequest.ignoreCustomErrors = true;
+        mySubmit();
+    }
```

Comment:
I thought it might be useful to ignore errors when pressing ctrl-enter, so this way we can force sending the request even when the custom model box reports an invalid custom model.